### ISSUE TITLE
Rewriting rpath fails when bundling ppa:beineri/opt-qt542 and earlier

### ISSUE THREAD

**Issue by probonopd** [probonopd/linuxdeployqt#31]
See these build artefacts:
https://github.com/eteran/edb-debugger/pull/458

Defect 1:
Qt 5.5 and 5.6 builds work, for the earlier builds some(!) of the libraries (namely, libxcb\* stuff from the system) are not correctly loaded from within the bundle.

Defect 2:
When trying to bundle Qt 4.x it seems to cripple the ELFs entirely...

Does anyone have a clue as to what causes these two defects?

**Comment by probonopd** [comment]
Might be related to https://github.com/NixOS/patchelf/issues/46 - @darealshinji do you have an idea?

**Comment by probonopd** [comment]
Duplicate of https://github.com/probonopd/linuxdeployqt/issues/29?

**Comment by probonopd** [comment]
Related to https://github.com/NixOS/patchelf/issues/99?

**Comment by probonopd** [comment]
Some binaries get patched correctly

```
patchelf --print-rpath squashfs-root/usr/lib/libQt5DBus.so.5
$ORIGIN
```

But some don't, for unknown reasons

```
find squashfs-root/usr/plugins/ -type f -exec patchelf --print-rpath {} \;
/opt/qt53/lib
/opt/qt53/lib
/opt/qt53/lib
/opt/qt53/lib
/opt/qt53/lib
/opt/qt53/lib
/opt/qt53/lib
```

Can be fixed with

```
find squashfs-root/usr/plugins/ -type f -exec patchelf --set-rpath '$ORIGIN/../../lib' {} \;
```

**Comment by probonopd** [comment]
Also, https://github.com/probonopd/linuxdeployqt/issues/79#issuecomment-289279675

When using https://launchpad.net/~beineri/+archive/ubuntu/opt-qt532-trusty then currently the following lands in the AppImage, which is clearly wrong:

```
[antergos@Antergos-ISO ~]$ strings squashfs-root/usr/lib/libQt5Core.so.5 | grep '^qt_.*='
qt_stngpath=/opt/qt53/etc/xdg
qt_prfxpath=..
qt_docspath=/opt/qt53/doc
qt_hdrspath=/opt/qt53/include
qt_libspath=/opt/qt53/lib
qt_lbexpath=/opt/qt53/libexec
qt_binspath=/opt/qt53/bin
qt_plugpath=plugins
qt_impspath=/opt/qt53/imports
qt_qml2path=/opt/qt53/qml
qt_adatpath=/opt/qt53
qt_datapath=/opt/qt53
qt_trnspath=/opt/qt53/translations
qt_xmplpath=/opt/qt53/examples
qt_tstspath=/opt/qt53/tests
qt_instdate=2015-03-11
qt_lcnsprod=OpenSource
qt_lcnsuser=Open Source
```

Needs to be fixed in https://github.com/probonopd/linuxdeployqt/issues/79#issuecomment-289279675

**Comment by probonopd** [comment]
Should have been fixed by fixing #79.
